src/eval.rs

```diff
@@ -252,56 +277,19 @@ fn eval_expr(expr: &Expr, env: &Env, program: &Program) -> Result<Value, Runtime
             let mut env_args = Env::new();
             let function = program.helpers.get(callee).unwrap().clone();
 
-            // check to see proper args
-            if args.len() != function.params.len() {
-                return Err(RuntimeError::IncorrectArgs);
-            }
-            let mut num_matches = 0; // checking to see if we've matched every arg
-
             for i in 0..args.len() {
-                let mut arg_val = eval_expr(args.get(i).unwrap(), env, program).unwrap();
-                println!("argval: {:?}", arg_val);
+                let mut arg_val = eval_expr(args.get(i).unwrap(), env, program)?;
                 let param_type = function.params.get(i).unwrap().clone().1;
-                println!("param_type: {:?} ", param_type);
-                let mut num_range = 0..0;
-                let mut num_type = false;
-                let mut num = 0;
-
-                let matches = match (&arg_val, param_type) {
-                    (Value::Bool(b), Type::BoolT) => true,
-                    (Value::Num(n, range), Type::NumT(t_range)) => {
-                        if range != &t_range {
-                            num_range = t_range.clone();
-                        } else {
-                            num_range = range.clone();
-                        }
-                        num_type = true;
-                        num = n.clone();
-                        true
+
+                arg_val = match (arg_val, param_type) {
+                    (Value::Num(n, _), Type::NumT(t_range)) => {
+                        cast(n, t_range, Overflow::Wraparound)?
```

Comment:
the typechecker guarantees that the types are correct. you shouldn't do any run-time type adjustment unless the semantics tell you to do so!

Reply:
so, remove this cast.